tower-buffer/src/worker.rs

```diff
@@ -28,6 +30,13 @@ where
     handle: Handle,
 }
 
+impl<T, R> Unpin for Worker<T, R>
```

Comment:
What's the argument for why this is okay?

 - `current_message`: do we guarantee that once we create a `Pin<Message>`, we never move it again?
 - `rx`: is `mpsc::Receiver` unconditionally `Unpin`?
 - `service`: do we ever create a `Pin` to the `Service`?
 - `finish`, `failed`, and `handle` are fine, since we never give out `Pin` to them

Reply:
I _think_ the argument for `current_message` is that we never give out a `Pin` to the message while it's sitting in `current_message`.